crates/node/builder/src/launch/engine.rs

```diff
@@ -191,15 +190,23 @@ where
             jwt_secret,
             engine_events: event_sender.clone(),
         };
-        let engine_payload_validator = add_ons.engine_validator(&add_ons_ctx).await?;
+        let validator_builder = add_ons.engine_validator_builder();
+
+        // First build the payload validator
+        let payload_validator = validator_builder.clone().build(&add_ons_ctx).await?;
+
+        // Build the engine validator with all required components
+        let engine_validator = validator_builder
+            .build_tree_validator(&add_ons_ctx, engine_tree_config.clone())
+            .await?;
 
         let consensus_engine_stream = UnboundedReceiverStream::from(consensus_engine_rx)
             .maybe_skip_fcu(node_config.debug.skip_fcu)
             .maybe_skip_new_payload(node_config.debug.skip_new_payload)
             .maybe_reorg(
                 ctx.blockchain_db().clone(),
                 ctx.components().evm_config().clone(),
-                engine_payload_validator.clone(),
+                payload_validator,
```

Comment:
oh I see we still need this here...

let's see if we can get rid of this

Reply:
ok, i've adapted the reorg functionality to use `EngineValidator` instead of `PayloadValidator`, ptal